src/AlarmClock.h

```diff
@@ -1,36 +1,46 @@
 /* 
  * File:   AlarmClock.h
- * Author: Craig Cogdill
- * Created: October 15, 2015 10:45am
+ * Author: Amanda Carbonari
+ * Created: January 5, 2016 4:35pm
+ * Description: Utilizes the boost library to create a singleton interruptable thread
+ *   that "communicates" with the main thread via a mutex, locks, and condition variables.
  */
 
 #pragma once
 #include <chrono>
 #include <thread>
+#include <mutex>
 #include <atomic>
-#include <future>
+#include <condition_variable>
 #include <functional>
+#include <iostream>
 #include "StopWatch.h"
-
+#include <cassert>
+using namespace std;
 
 template<typename Duration> class AlarmClock {
 public:
-   typedef std::chrono::microseconds microseconds;
-   typedef std::chrono::milliseconds milliseconds;
-   typedef std::chrono::seconds seconds;
-   
-   AlarmClock(unsigned int sleepDuration, std::function<void(unsigned int)> funcPtr = nullptr) : mExpired(false),
+   typedef chrono::microseconds microseconds;
+   typedef chrono::milliseconds milliseconds;
+   typedef chrono::seconds seconds;
+
+   AlarmClock(unsigned int sleepDuration, function<void (unsigned int)> funcPtr = nullptr) : mExpired(false),
+      mSleptTime(0),
+      mExit(false),
+      mReset(false),
+      mConstructorThreadId(this_thread::get_id()),
       kSleepTime(sleepDuration),
       kSleepTimeMsCount(ConvertToMillisecondsCount(Duration(sleepDuration))),
       kSleepTimeUsCount(ConvertToMicrosecondsCount(Duration(sleepDuration))),
       mSleepFunction(funcPtr) {
          if (mSleepFunction == nullptr) {
-            mSleepFunction = std::bind(&AlarmClock::SleepUs, this, std::placeholders::_1);
```

Comment:
Please use a lambda expression instead. It should have less overhead than going through a std::bind
```
mSleepFunction = [&](unsigned int sleepTime) 
{ 
  SleepUs(sleepTime); 
}; 
```